sdk/inc/azure/core/internal/az_hfsm_pipeline_internal.h

```diff
@@ -0,0 +1,180 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file az_hfsm_pipeline.h
+ * @brief Definition of #az_hfsm_pipeline and related types describing a bi-directional HFSM
+ *        pipeline.
+ *
+ * @remarks Both a non-blocking I/O (default) as well as a blocking I/O implementations are
+ * available. To enable the blocking mode, set TRANSPORT_MQTT_SYNC.
+ *
+ * @note The blocking I/O model (when TRANSPORT_MQTT_SYNC is defined) is not thread-safe.
+ */
+
+#ifndef _az_HFSM_PIPELINE_H
+#define _az_HFSM_PIPELINE_H
+
+// Define AZ_HFSM_PIPELINE_SYNC to manually control when pipeline events are executed.
+// The default action is to immediately process events on the same stack.
+// #define AZ_HFSM_PIPELINE_SYNC
+
+#include <azure/core/internal/az_hfsm_internal.h>
+#include <azure/core/az_platform.h>
+#include <azure/core/az_result.h>
+#include <stdint.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+/**
+ * @brief Used to declare HFSM policies.
+ *
+ */
+
+typedef struct az_hfsm_policy az_hfsm_policy;
+
+typedef struct az_hfsm_pipeline az_hfsm_pipeline;
```

Comment:
Missing documentation.
Also, I think in the end, we will need to consider exposing the policy as public and maintain the pipeline as private.

Reply:
Will add documentation. Will expose it as public in following PR.

Reply:
There is already documentation for az_hfsm_policy and az_hfsm_pipeline in the declarations.